Foundation/URLSession/http/EasyHandle.swift

```diff
@@ -170,6 +171,9 @@ extension _EasyHandle {
         let protocols = (CFURLSessionProtocolHTTP | CFURLSessionProtocolHTTPS)
         try! CFURLSession_easy_setopt_long(rawHandle, CFURLSessionOptionPROTOCOLS, protocols).asError()
         try! CFURLSession_easy_setopt_long(rawHandle, CFURLSessionOptionREDIR_PROTOCOLS, protocols).asError()
+        errorBuffer.withUnsafeMutableBufferPointer {
```

Comment:
`setAllowedProtocolsToHTTPAndHTTPS()` doesn't seem to be an accurate name for this function any more.

Reply:
I’m just using that function for some other initialisation as a convenience rather than create a new one. I think the name if probably good as it is as it relates to it’s primary purpose.

Reply:
I could move this extra code into another of the setup functions.

Reply:
What do you think @pushkarnk ?

Reply:
I see there is a function set(errorBuffer:) which is called from _HTTPURLProtocolconfigureEasyHandle(forRequest:) I could set the buffer there but it wouldn’t guarantee the buffer still exists when you call curl_easy_strerror() on the handle.

Reply:
I’ve pushed a new revision introducing a default error buffer so hte messages are always as descriptive as possible. See what you think.